lte/gateway/c/oai/tasks/s1ap/s1ap_mme.c

```diff
@@ -132,22 +130,17 @@ static int handle_message(zloop_t* loop, zsock_t* reader, void* arg) {
        * New message received from SCTP layer.
        * * * * Decode and handle it.
        */
-      s1ap_message message = {0};
+      S1ap_S1AP_PDU_t pdu = {0};
 
       // Invoke S1AP message decoder
-      if (s1ap_mme_decode_pdu(
-              &message, SCTP_DATA_IND(received_message_p).payload,
-              &message_id) < 0) {
+      if (s1ap_mme_decode_pdu(&pdu, SCTP_DATA_IND(received_message_p).payload) <
+          0) {
         // TODO: Notify eNB of failure with right cause
         OAILOG_ERROR(LOG_S1AP, "Failed to decode new buffer\n");
       } else {
         s1ap_mme_handle_message(
             state, SCTP_DATA_IND(received_message_p).assoc_id,
-            SCTP_DATA_IND(received_message_p).stream, &message);
-      }
-
-      if (message_id != MESSAGES_ID_MAX) {
-        s1ap_free_mme_decode_pdu(&message, message_id);
```

Comment:
Where/how is `pdu` getting freed?

Reply:
pdu variable is allocated in the procedure on stack  so does not require freeing.